src/components/common/toast/toast.tsx

```diff
@@ -0,0 +1,67 @@
+import { Toast } from "@lifesg/react-design-system/toast";
+import { useEffect, useRef } from "react";
+import { EToastTypes, IToast, useDisplay } from "src/context-providers";
+
+interface IProps {
+    toast: IToast;
+    toastFunction?: () => void;
+}
+
+export const DisplayToast = ({ toast, toastFunction }: IProps) => {
+    // =========================================================================
+    // CONST, STATE, REF
+    // =========================================================================
+    const { dismissToast } = useDisplay();
+    const timeoutRef = useRef(null);
+
+    // =============================================================================
+    // HELPER FUNCTIONS
+    // =============================================================================
+    function startDismissTimer(id: string) {
+        return setTimeout(() => {
+            dismissToast(id);
+        }, 4500);
+    }
+
+    // =========================================================================
+    // EFFECTS
+    // =========================================================================
+    useEffect(() => {
+        timeoutRef.current = startDismissTimer(toast.id);
+
+        return () => {
+            clearTimeout(timeoutRef.current);
+        };
+    }, [toast, dismissToast]);
+
+    // =============================================================================
+    // RENDER FUNCTIONS
+    // =============================================================================
+    const renderToast = () => {
+        switch (toast.type) {
+            case EToastTypes.SUCCESS_TOAST:
+            case EToastTypes.DELETE_TOAST: {
+                return (
+                    <Toast
+                        type={"success"}
+                        label={toast.message}
+                        autoDismiss
+                        fixed={false}
+                    />
+                );
+            }
```

Comment:
no need to wrap the case block in brackets because there are no variables to keep within the block
```suggestion
            case EToastTypes.SUCCESS_TOAST:
            case EToastTypes.DELETE_TOAST:
                return (
                    <Toast
                        type={"success"}
                        label={toast.message}
                        autoDismiss
                        fixed={false}
                    />
                );
```

Reply:
changes are made on the latest push